File: EmployeeManagement/Models/LeaveRequest.cs
namespace EmployeeManagement.Models;

public class LeaveRequest
{
    public Guid Id { get; set; }

    public Guid EmployeeId { get; set; }

    public Guid LeaveTypeId { get; set; }

    public DateTime StartDate { get; set; }

    public DateTime EndDate { get; set; }

    public int Days { get; set; }

    public LeaveStatus Status { get; set; }

    public string? Reason { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? ReviewedAt { get; set; }

    public Guid? ReviewedBy { get; set; }

    public Employee Employee { get; set; } = null!;

    public LeaveType LeaveType { get; set; } = null!;

    public bool IsDeleted { get; set; } = false;
}

File: employee-management-client/src/types/leaveRequest.ts
export type LeaveStatus =
    | "Pending"
    | "Approved"
    | "Rejected";

export interface LeaveRequest {
    id: string;
    leaveTypeId: string;
    leaveTypeName: string;
    startDate: string;
    employeeCode:string;
    employeeName:string;
    employeEmail:string;
    endDate: string;
    days: number;
    status: LeaveStatus;
    createdAt: string;
    reviewedAt?: string | null;
    reason:string | undefined;
    reviewedBy?: string | null;
}

export interface LeaveRequestCreateRequest {
    leaveTypeId: string;
    startDate: string;
    reason:string | undefined;
    endDate: string;
}

export interface LeaveBalance {
    leaveTypeId: string;
    leaveTypeName: string;
    annualLimit: number;
    consumedDays: number;
    pendingDays: number;
    remainingDays: number;
}

export interface PaginationQuery {
    page?: number;
    pageSize?: number;
    search?: string;
}

export interface PagedResult<T> {
    items: T[];
    page: number;
    pageSize: number;
    totalCount: number;
}